### PR TITLE
Number parsing: Allow to specify `groupmark`

### LINKED ISSUE #626: Ability to specify a thousands separator character
This feature was listed in https://github.com/JuliaData/CSV.jl/issues/3 , but not implemented.  

https://github.com/JuliaData/CSV.jl/issues/207 has a complicated workaround (but keyword transforms does not exists anymore?), and says "there isn't a "thousands" separator option, but we could potentially add on".

Practical example, data is two first rows from table 2 on page 10 in https://www.istat.it/it/files//2020/05/Rapporto_Istat_ISS.pdf
```
a=CSV.File(IOBuffer("""
       Alessandria 95,7 98,2 -12,8 91,0 1.199 693 222 18,5
       Ancona 76,6 84,3 -10,7 49,4 704 528 86 12,2
       """), delim=" ", header = false, decimal = ',')
```
Row 1, column 6 would ideally be Int 1199.   Here comma is a decimal separator and period is a thousands separator.

### PR DESCRIPTION
**Pull request by barucden**
This optional argument can be used to parse numbers containing thousand separators.

As suggested in the issue referenced below, the implementation just sets `groupmark` in the `Parsers.Options` constructor.

Resolves #626

### CODE REVIEW

**Comment by bkamins** [comment]
Thank you for submitting a PR :)

File: src/context.jl [hunk 1]
```diff
@@ -245,7 +246,7 @@ function Context(source::ValidSources;
...
     validate::Bool=true,
     )
-    return @refargs Context(source, header, normalizenames, datarow, skipto, footerskip, transpose, comment, ignoreemptyrows, ignoreemptylines, select, drop, limit, buffer_in_memory, threaded, ntasks, tasks, rows_to_check, lines_to_check, missingstrings, missingstring, delim, ignorerepeated, quoted, quotechar, openquotechar, closequotechar, escapechar, dateformat, dateformats, decimal, truestrings, falsestrings, stripwhitespace, type, types, typemap, pool, downcast, lazystrings, stringtype, strict, silencewarnings, maxwarnings, debug, parsingdebug, validate, false)
+    return @refargs Context(source, header, normalizenames, datarow, skipto, footerskip, transpose, comment, ignoreemptyrows, ignoreemptylines, select, drop, limit, buffer_in_memory, threaded, ntasks, tasks, rows_to_check, lines_to_check, missingstrings, missingstring, delim, ignorerepeated, quoted, quotechar, openquotechar, closequotechar, escapechar, dateformat, dateformats, decimal, grouopmark, truestrings, falsestrings, stripwhitespace, type, types, typemap, pool, downcast, lazystrings, stringtype, strict, silencewarnings, maxwarnings, debug, parsingdebug, validate, false)
```
Comment (nickrobinson251) [inline src/context.jl:249]: typo
```suggestion
    return @refargs Context(source, header, normalizenames, datarow, skipto, footerskip, transpose, comment, ignoreemptyrows, ignoreemptylines, select, drop, limit, buffer_in_memory, threaded, ntasks, tasks, rows_to_check, lines_to_check, missingstrings, missingstring, delim, ignorerepeated, quoted, quotechar, openquotechar, closequotechar, escapechar, dateformat, dateformats, decimal, groupmark, truestrings, falsestrings, stripwhitespace, type, types, typemap, pool, downcast, lazystrings, stringtype, strict, silencewarnings, maxwarnings, debug, parsingdebug, validate, false)
```
sorry there doesn't seem to be an existing test for this method -- would you mind adding one? I think even just `@test Context("") isa Context` would have caught this.

**Comment by barucden** [comment]
I added validation.

**Comment by barucden** [comment]
Bump :)

**Comment by barucden** [comment]
As far as I can tell, the very same functionality has been implemented and merged in #1093. Closing this one...

**Pull request closed without merging**